### PR TITLE
New features for pregnant plugin

### PR DESCRIPTION
**Pull request by XspeedPL**
New features:
- pregnancy roll bonus on repeated internal shots
- switch to pregnant sprites after repeated internal shots
- dripping in non-penetrative states

Also changed default fertility to 30% to facilitate the use of the new bonus.

### CODE REVIEW

**Comment by XspeedPL** [comment]
FYI, VS Code applied lots of automatic whitespace changes to JSON files, to avoid looking through them on GitHub you can use the `w=1` URL switch

https://github.com/ErinaSugino/Starbound-SBR-Mod/pull/22/files?w=1

File: scripts/sexbound/plugins/pregnant/pregnant.lua [hunk 1]
```diff
@@ -834,6 +923,8 @@ end
 
 --- Validates the loaded config and sets missing config options to be default values.
 function Sexbound.Actor.Pregnant:validateConfig()
+    self:validateDripTimer(self._config.dripTimer)
```
Comment (ErinaSugino) [inline scripts/sexbound/plugins/pregnant/pregnant.lua:926]: Having an option to configure the drip decay timer is a good idea - however, there is no actual config value added that represents this.

File: scripts/sexbound/plugins/pregnant/pregnant.lua [hunk 2]
```diff
@@ -157,6 +205,8 @@ function Sexbound.Actor.Pregnant:handleBecomePregnant(message)
             
             self:getLog():debug("Pregnancy check for actor "..self:getParent():getActorNumber()..": Allow species " .. tostring(compatible) .. " - Can Impregnate " .. tostring(canImpregnate) .. " - Can be impregnated " .. tostring(canPregnate) .. " - Insertion " .. tostring(directInsertion))
 
+            self:handleInsemination(actor)
```
Comment (ErinaSugino) [inline scripts/sexbound/plugins/pregnant/pregnant.lua:208]: Is this timed right?

File: scripts/sexbound/plugins/pregnant/pregnant.lua [hunk 3]
```diff
@@ -40,10 +40,58 @@ function Sexbound.Actor.Pregnant:onMessage(message)
     end
 end
 
+--- Returns a reference to this actor's insemination decay rate.
+function Sexbound.Actor.Pregnant:getInseminationDecay()
```
Comment (ErinaSugino) [inline scripts/sexbound/plugins/pregnant/pregnant.lua:44]: This should be configurable too.

File: scripts/sexbound/plugins/pregnant/pregnant.lua (hunk 1 above)
Comment (ErinaSugino) [reply scripts/sexbound/plugins/pregnant/pregnant.lua:926]: Yes, the thing is *I* can set it. But a user should never have to try and edit the code themselves. They already implode their game enough trying to edit the config.

I'm sure a setting for this specific part is not necessary, but it also wouldn't surprise me when people start asking rather sooner than later if they can adjust the values, so I brought it up as a pointer.

File: scripts/sexbound/plugins/pregnant/pregnant.lua (hunk 3 above)
Comment (ErinaSugino) [reply scripts/sexbound/plugins/pregnant/pregnant.lua:44]: The only two things you added to the config are the multiplier and max value. Nothing regarding the decay rate. So you'll have to include that too.

**Comment by ErinaSugino** [comment]
On another note:

The mechanic consists of two independent parts - the added fertility bonus on successful sub-sequent climaxes, as well as the cumflation logic of displaying the pregnant belly sprite after enough overall climaxes. So far, however, both these mechanics are triggered simultaneously resulting in a logical error.

That the fertility bonus is triggered after all pregnancy validation checks ran makes sense. That, however, the general "you're getting full" part of the logic too is bound by (except for the actual fertility roll) all requirements for a pregnancy being fulfilled is not.

Here is a good example: SBR only sends a pregnancy attempt request from the climax plugin to the pregnancy plugin when the climaxing actor has the necessary conditions to impregnate anyone in the first place. However, different conditions are applied to generally climaxing. This results in, e.g., an infertile or sterilized actor still logically climaxing, still logically "filling the other actor up", but never sending a request to the pregnancy plugin, hence never increasing the statistic.

You should probably outsource this part of the logic elsewhere, or at least make sure the actual tracking of current stored shots is triggered independently from impregnation requests, by making the climax plugin send another different event message you hook into.

File: scripts/sexbound/plugins/pregnant/pregnant.lua [hunk 4]
```diff
@@ -834,6 +923,8 @@ end
 
 --- Validates the loaded config and sets missing config options to be default values.
 function Sexbound.Actor.Pregnant:validateConfig()
+    self:validateDripTimer(self._config.dripTimer)
+    self:validateInseminations(self._config.inseminations)
```
[the remark this answers is not in the document]
Comment (ErinaSugino) [reply scripts/sexbound/plugins/pregnant/pregnant.lua]: Fixed as of ff567c3

**Review by ErinaSugino — APPROVED** [review]
Good enough. The fertility bonus vs. inflation split still needs more solid separation, though this isn't critical for now. I'll look into this later.

**Comment by ErinaSugino** [comment]
No need to make it its own plugin. But like I already mentioned in my previous notes, the climax plugin will have be involved in some way.

**Pull request merged**